Watchman.Discord/Areas/Users/Controllers/WarnsController.cs

```diff
@@ -43,10 +44,19 @@ public async Task AddWarn(AddWarnCommand command, Contexts contexts)
             await messageService.SendResponse(x => x.UserHasBeenWarned(contexts.User.Name, mentionedUser.Name, command.Reason));
         }
 
-        public async Task Warns(WarnsCommand command, Contexts contexts)
+        public Task Warns(WarnsCommand command, Contexts contexts)
         {
-            var mentionedUser = command.User == 0 ? contexts.User : await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
-            await this._warnService.GetWarns(command, contexts, mentionedUser, contexts.Server.Id);
+            var messageService = _messagesServiceFactory.Create(contexts);
+            var mentionedUser = command.User == 0 ? 
+                    contexts.User 
+                        : 
+                    this._usersService.GetUserByIdAsync(contexts.Server, command.User).GetAwaiter().GetResult();
```

Comment:
dlaczego `.GetAwaiter().GetResult()`?
lepiej po prostu await

Reply:
GetAwaiter().GetResult() jest synchroniczne, z await musiałbym zrobić metodę Warns() jako async, a na dole metody zamiast return messageService(...) zrobić awaity

a zależy mi żeby Warns nie było async tylko zwracało Task, chyba że definitywnie mam to pozmieniać na awaity ale mi się wydaje że tak jest wydajniej

Reply:
GetAwaiter().GetResult() wygląda bardzo dziwnie i nigdzie tego w projekcie nie używamy
odkąd jest await to raczej stosuje się await
Co do wydajności, to ok - jeśli można osiągnąć lekko większą wydajność bez utraty na jakości kodu i dużego wysiłku
ale w tym przypadku, różnica w wydajności między await a return pewnie sięga mniej niż 1ms, czy nic, a jakość kodu na tym cierpi